### PR TITLE
procd: add build date information to 'ubus call system board'

### PR DESCRIPTION
**Pull request by feckert**
This pullrequest adds the necessary information to the ` build system` so that the build time of the firmware can be retrieved via the `ubus call system board` call.

```
{
        "kernel": "6.6.36",
        "hostname": "G3-10940",
        "system": "AMD GX-412TC SOC",
        "board_name": "pc-engines-apu3",
        "rootfs_type": "squashfs",
        "board_version": "unknown",
        "release": {
               ....
                "revision": "r0+27157-e8661255fe",
                "target": "x86/64",
--->            "builddate": "1721382861",
                "image_name": "TDT-LW"
        }
}

```

The time information used for this comes from the build system variable `SOURCE_DATE_EPOCH`. This information is read out with the [get_source_date_epoch](
https://github.com/openwrt/openwrt/blob/master/scripts/get_source_date_epoch.sh#L13) script during compilation. This information is normally the last commit date in the openwrt repository. This information is used to make the **reproducible builds**. 

I already [posted](https://patchwork.ozlabs.org/project/openwrt/patch/20240620104555.24233-1-fe@dev.tdt.de/) the preparation patch for `procd` to read the information from `/usr/lib/os-release` instead of `/etc/openwrt_release` on the mailing list a month ago. Unfortunately I probably didn't trigger the right people to look at it for the `procd`, since the big picture was probably missing?

https://github.com/openwrt/procd/pull/11

### CODE REVIEW

File: package/system/procd/patches/0012-use-os-release.patch [hunk 1]
```diff
@@ -0,0 +1,57 @@
+From 51c2d32c98b2ae92d00820c3734dec525766b1fa Mon Sep 17 00:00:00 2001
+From: Florian Eckert <fe@dev.tdt.de>
+Date: Tue, 18 Jun 2024 15:05:45 +0200
+Subject: [PATCH 1/2] system: replace openwrt_release with os-release
```
Comment (ynezz) [inline package/system/procd/patches/0012-use-os-release.patch:4]: FYI there is https://github.com/openwrt/procd where this belongs.

Comment (feckert) [reply package/system/procd/patches/0012-use-os-release.patch:4]: Is procd now maintained on github?
I was not aware of that.
I will open a PR asap to procd
Thanks :+1:

Comment (feckert) [reply package/system/procd/patches/0012-use-os-release.patch:4]: If the development is now on github, then we should remove in the repository description on github
the **MIRROR** `-->[MIRROR]<-- OpenWrt service / process manager`.

Comment (KanjiMonster) [reply package/system/procd/patches/0012-use-os-release.patch]: > If the development is now on github, then we should remove in the repository description on github the **MIRROR** `-->[MIRROR]<-- OpenWrt service / process manager`.

It works the same way as this repository. You can create PRs, but they won't get merged via github and instead via the repo on git.openwrt.org.

Comment (KanjiMonster) [reply package/system/procd/patches/0012-use-os-release.patch]: What @ynezz mainly wanted to express is that you shouldn't add patches for OpenWrt projects in OpenWrt itself, and instead submit the changes directly to the appropriate sub project.

Whether this happens via a PR on github or via patches to the mailing list is secondary.

**Comment by hauke** [comment]
Thank you for the PR, applied to main in 6e6b82ac5f2a33ec2bf14a3d0ac2f03fd787282b.

**Pull request closed without merging**